### DISCUSSION TITLE
Shared presets file is ignored

### DISCUSSION THREAD

**Discussion by nosajsnikta** [renovatebot/renovate#8457] [Request Help] [answered]
I'd like to central our renovate configuration into a single presets file for a number of projects stored in Gitlab
* We are using an on-premise GitLab server
* We have 5 different projects, each with the same contents in their renovate.json

I have followed the help and 
* Created new GitLab repo called <hosted-gitlab-server>/my.name/renovate-config
* Added a single renovate.json file into that repo containing the following:
```
{
  "name": "renovate-config-projectx",
  "version": "0.0.1",
  "renovate-config": {
    "default": {
      "extends": ["config:base", "docker:enableMajor", ":semanticCommitTypeAll(fix)"],
      "automerge": true,
      "major": {
        "automerge": false
      },
      "gomod": {
        "postUpdateOptions": ["gomodTidy"]
      }
    }
  }
}
```

I have then replaced the contents of the renovate.json in one of the projects with:
```
{
  "extends": ["local>my.name/renovate-config"]
}

```

When I run renovate, it doesn't appear to apply the presets defined in my central renovate.json.
* I know the central renovate.json is being read, because if I put a syntax error in it, my Jenkins job that runs renovate fails due to that

Any ideas what I could have missed out?

**Comment by rarkins** [comment, marked as answer]
Convert it to a plain JSON config like this instead:

```json
{
      "extends": ["config:base", "docker:enableMajor", ":semanticCommitTypeAll(fix)"],
      "automerge": true,
      "major": {
        "automerge": false
      },
      "gomod": {
        "postUpdateOptions": ["gomodTidy"]
      }
}
```

**Reply by nosajsnikta** [reply]
Ok thanks again I will retry with this format

**Reply by nosajsnikta** [reply]
Thanks once again @rarkins...all seems good now

**Comment by viceice** [comment]
> ```
> {
>   "name": "renovate-config-projectx",
>   "version": "0.0.1",
>   "renovate-config": {
>     "default": {
>       "extends": ["config:base", "docker:enableMajor", ":semanticCommitTypeAll(fix)"],
>       "automerge": true,
>       "major": {
>         "automerge": false
>       },
>       "gomod": {
>         "postUpdateOptions": ["gomodTidy"]
>       }
>     }
>   }
> }
> ```

@nosajsnikta Where did you found this sample?

**Reply by nosajsnikta** [reply]
Well it was the final resting point I got to before I had to switch to some other work. I originally started out with what @rarkins has said but that didn't seem to work so it became a bit of try/see exercise hence it might be a bit garbled

Primary source of info was the help at https://docs.renovatebot.com/config-presets/#local-presets and then
1. Section "How to Publish Preset Configs"
2. Section "GitLab-hosted Presets"

**Reply by viceice** [reply]
Thanks. That section needs rework.

@HonkingGoose Can you refactor the publish sample, as that is only right for npm package publishing, not for other shared presets.

@rarkins Maybe deprecate (docs only) npm presets and move to seperate section at bottom?

**Reply by HonkingGoose** [reply]
> @HonkingGoose Can you refactor the publish sample, as that is only right for npm package publishing, not for other shared presets.

I'm not sure what you want me to do.
Can you open a new issue to discuss this in more detail please?
